### DISCUSSION TITLE
What's required to reference a WinUI 3 UserControl project in Uno Platform?

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [unoplatform/uno#13292] [Q&A] [answered]
I'm new to Uno Platform.  I'm using Visual Studio 2022 (17.8, preview).
I want to use several WinUI 3 UserControls I've developed in an Uno Platform solution. 
Within the solution, the desired UserControl project successfully rebuilds, but when I attempt to reference this WinUI 3 Class Library (WinUI 3 in Desktop) within an Uno Platform project, I unhappily get yellow triangles and the application will not compile or deploy.

Is there any documentation providing guidance for referencing Class Libraries in an Uno Platform project?
Is this even possible? Or is extra work required to bring a .NET 7 / WindowsAppSDK Class Library into the Uno Platform universe?

**Comment by limeniye** [comment, marked as answer]
Hello, @MarkBrainfire, and thank you for your report.
To fix this problem do not use the **WinUI 3 Library**. 
Use the **Uno Platform Library**

![image](https://github.com/unoplatform/uno/assets/69314237/fecd2c04-f2ae-4a62-ab3f-98df987f5ff6)

You can see an example there:
https://github.com/unoplatform/uno/discussions/13065

**Reply by MarkBrainfire** [reply]
Thanks limeniye,

Yes, I now realize that a Uno Project can reference a .net class library just fine, but cannot reference a WinUI 3 class library, but rather must use an Uno Platform Library.

I'm now developing a utility to overcome this that will automatically convert a WinUI 3 application to a Uno Platform Application and a WinUI 3 Class Library to an Uno Platform Class Library.  Anyone that wants to help me with this project is welcome.
